acceptance/openstack/sharedfilesystems/v2/shareaccessrules.go

```diff
@@ -16,3 +18,37 @@ func ShareAccessRuleGet(t *testing.T, client *gophercloud.ServiceClient, accessI
 
 	return accessRule, nil
 }
+
+func WaitForShareAccessRule(t *testing.T, client *gophercloud.ServiceClient, accessRule *shareaccessrules.ShareAccess, status string) error {
+	if accessRule.State == status {
+		return nil
+	}
+
+	return tools.WaitFor(func() (bool, error) {
+		latest, err := ShareAccessRuleGet(t, client, accessRule.ShareID)
+		if err != nil {
```

Comment:
If we expect delay when creating the share access rule, we should not treat a 404 as an error case: 
https://github.com/gophercloud/gophercloud/actions/runs/3566566178/jobs/5993171793#step:5:1098